src/Compilers/CSharp/Portable/Parser/LexerCache.cs

```diff
@@ -30,25 +31,81 @@ internal partial class LexerCache
                                 return kind;
                             });
 
-        private readonly TextKeyedCache<SyntaxTrivia> _triviaMap;
-        private readonly TextKeyedCache<SyntaxToken> _tokenMap;
-        private readonly CachingIdentityFactory<string, SyntaxKind> _keywordKindMap;
+        private TextKeyedCache<SyntaxTrivia> _triviaMap;
+        private TextKeyedCache<SyntaxToken> _tokenMap;
+        private CachingIdentityFactory<string, SyntaxKind> _keywordKindMap;
         internal const int MaxKeywordLength = 10;
 
-        internal LexerCache()
+        private PooledStringBuilder _stringBuilder;
+        private readonly char[] _identBuffer;
+        private SyntaxListBuilder _leadingTriviaCache;
+        private SyntaxListBuilder _trailingTriviaCache;
+
+        private const int LeadingTriviaCacheInitialCapacity = 128;
+        private const int TrailingTriviaCacheInitialCapacity = 16;
+
+        private LexerCache()
+        {
+            _identBuffer = new char[32];
+
+            Initialize();
```

Comment:
>Initialize(); [](http://example.com/codeflow?start=12&length=13)

Is this initialization redundant?

Reply:
Let me see what changing the members to nullable would look like

Reply:
@AlekseyTs -- Done in commit 3. It's more verbose, but probably a bit easier to understand as it doesn't require the Initialize code, but instead defers the member population.